fe/fe-core/src/main/java/org/apache/doris/statistics/StatisticsJobManager.java

```diff
@@ -18,56 +18,128 @@
-    // Rule3: The job for external table is not supported
-    private void checkRestrict(Set<Long> tableIdList) {
-        // TODO
+    /**
+     * Rule1: The same table cannot have two unfinished statistics jobs
+     * Rule2: The unfinished statistics job could not more then Config.max_statistics_job_num
+     * Rule3: The job for external table is not supported
+     */
+    private void checkRestrict(long dbId, Set<Long> tableIds) throws DdlException {
+        Database db = Catalog.getCurrentCatalog().getDbOrDdlException(dbId);
+
+        // check table type
+        for (Long tableId : tableIds) {
+            Table table = db.getTableOrDdlException(tableId);
+            if (table.getType() != Table.TableType.OLAP) {
+                throw new DdlException("The external table(" + table.getName() + ") is not supported.");
+            }
+        }
+
+        int unfinishedJobs = 0;
+        StatisticsJobScheduler jobScheduler = Catalog.getCurrentCatalog().getStatisticsJobScheduler();
+        Queue<StatisticsJob> statisticsJobs = jobScheduler.pendingJobQueue;
+
+        // check table unfinished job
+        for (StatisticsJob statisticsJob : statisticsJobs) {
+            JobState jobState = statisticsJob.getJobState();
+            List<Long> tableIdList = statisticsJob.getTableIds();
+            if (jobState == JobState.PENDING || jobState == JobState.SCHEDULING || jobState == JobState.RUNNING) {
+                for (Long tableId : tableIds) {
+                    if (tableIdList.contains(tableId)) {
+                        throw new DdlException("The same table(id=" + tableId + ") cannot have two unfinished statistics jobs.");
+                    }
+                }
+                unfinishedJobs++;
+            }
+        }
+
+        // check the number of unfinished tasks
+        if (unfinishedJobs > Config.cbo_max_statistics_job_num) {
+            throw new DdlException("The unfinished statistics job could not more then Config.cbo_max_statistics_job_num.");
+        }
     }
 
-    private void checkPermission() {
-        // TODO
+    private void checkPermission(long dbId, Set<Long> tableIds, UserIdentity userInfo) throws DdlException {
+        PaloAuth auth = Catalog.getCurrentCatalog().getAuth();
+        Database db = Catalog.getCurrentCatalog().getDbOrDdlException(dbId);
+
+        // check the db permission
+        String dbName = db.getFullName();
+        boolean dbPermission = auth.checkDbPriv(userInfo, dbName, PrivPredicate.SELECT);
+        if (!dbPermission) {
+            throw new DdlException("You do not have permissions to analyze the database(" + dbName + ").");
+        }
+
+        // check the tables permission
+        for (Long tableId : tableIds) {
+            Table tbl = db.getTableOrDdlException(tableId);
+            boolean tblPermission = auth.checkTblPriv(userInfo, dbName, tbl.getName(), PrivPredicate.SELECT);
```

Comment:
Is it possible to check the table directly for permission verification? 
Does having the select permission of db mean that it must have the select permission of all tables in db?

Reply:
yes, If it is all tables, then query the permissions of the db, otherwise only query the permissions of the table.